src/main/java/com/example/purerestapi/controller/UserController.java

```diff
@@ -0,0 +1,89 @@
+package com.example.purerestapi.controller;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+import com.example.purerestapi.controller.assembler.NoteAssembler;
+import com.example.purerestapi.controller.assembler.UserAssembler;
+import com.example.purerestapi.entity.Note;
+import com.example.purerestapi.entity.User;
+import com.example.purerestapi.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.IanaLinkRelations;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Controller
+@ResponseBody
```

Comment:
You can use '@RestController' instead that is itself annotated with @Controller and @ResponseBody.

Reply:
Good point, I have just done it as a test, and one more time show @RestController it is easier and more understandable than just annotation